### PR TITLE
Added Validations for Team endpoints

### PR DESCRIPTION
**Pull request by 1mac11**

### CODE REVIEW

File: core/utils/checkers.py [hunk 1]
```diff
@@ -1,6 +1,59 @@
...
+
+def check_size(workers):
+    if workers.count() > 5:
+        raise ValidationError({'size': "Team should be consist of no more than 5 workers."})
+
+
+def update_team_check(team: Team, workers, managers):
+    workers = team.users.filter(detail__type__name=UserTypeChoices.WORKER) | workers
+    managers = team.users.filter(detail__type__name=UserTypeChoices.MANAGER) | managers
+    check_managers(managers, team)
```
Comment (ahodkov) [inline core/utils/checkers.py]: update and create has the same code, maybe, do you can add a new function for this?

File: core/utils/checkers.py [hunk 2]
```diff
@@ -1,6 +1,59 @@
...
+            if manager.teams.exists():
+                raise ValidationError({'manager': "Manager shouldn't be on other teams"})
+    else:
+        if managers.exclude(Q(teams__isnull=True) | Q(teams=team)).exists():
+            raise ValidationError({'manager': "Manager shouldn't be on other teams"})
+
+
+def check_workers(workers, team=None):
+    if team is None:
+        for worker in workers:
```
Comment (ahodkov) [inline core/utils/checkers.py]: Also, you can use function:
```
def check_user(user, type_user):
    for user in users:
       if user.team.exists():
           raise ValidationError({type_user: f"{type_user} shouldn't be on other teams"})
```

**Review by ahodkov — CHANGES_REQUESTED** [review]
Better to use db optimization, because, for example, in update_team_check we call user at least twice(if team has 2 users) and call so many times for team database.

**Review by ahodkov — APPROVED** [review]

**Comment by Zarif99** [comment]
merge

**Pull request merged**